### PR TITLE
Move queries with variable number of args to a fixed number of args syntax

### PR DESCRIPTION
**Pull request by ftkg**
Queries with a variable number of arguments are treated as distinct queries from PostgreSQL perspective, and their statistics are tracked separately. This PR aims to improve query metrics' cardinality and, consequently, improve query stats monitoring. This is achieved by using a fixed syntax:

- `WHERE x in ($1,$2,$3, ...)` is replaced with `WHERE x = ANY($1::type_of_x[])`, with all values passed as a single array argument.
- `INSERT INTO t (a,b,c) VALUES ($1,$2,$3), ($4,$5,$6), ...` is replaced with `pgx.Batch`, with a single row insert per query.

### CODE REVIEW

File: server/console_channel.go [hunk 1]
```diff
@@ -66,20 +64,16 @@ func (s *ConsoleServer) DeleteChannelMessages(ctx context.Context, in *console.D
...
+		for _, id := range in.Ids {
+			_, err := uuid.FromString(id)
 			if err != nil {
 				return nil, status.Error(codes.InvalidArgument, "Requires a valid message ID.")
 			}
-			params = append(params, idStr)
-			statements[i] = "$" + strconv.Itoa(i+1)
 		}
-		query := "DELETE FROM message WHERE id IN (" + strings.Join(statements, ",") + ")"
+		query := "DELETE FROM message WHERE id = ANY($1)"
```
Comment (redbaron) [inline server/console_channel.go:73]: does it work with generic `database/sql` interface without converting to `pgx`?

File: server/core_wallet.go [hunk 2]
```diff
@@ -216,8 +208,7 @@ func updateWallets(ctx context.Context, logger *zap.Logger, tx pgx.Tx, updates [
 				return nil, err
 			}
 
-			params = append(params, uuid.Must(uuid.NewV4()), userID, changesetData, update.Metadata)
-			statements = append(statements, fmt.Sprintf("($%v::UUID, $%v, $%v, $%v)", strconv.Itoa(len(params)-3), strconv.Itoa(len(params)-2), strconv.Itoa(len(params)-1), strconv.Itoa(len(params))))
+			batchLedger.Queue(queryLedger, uuid.Must(uuid.NewV4()), userID, changesetData, update.Metadata)
```
Comment (redbaron) [inline server/core_wallet.go:211]: `uuid.Must` seems unnecessary becasue `uuid.NewV4()` doesn't fail

File: server/console_channel.go (hunk 1 above)
Comment (ftkg) [reply server/console_channel.go:73]: Yes, this seemingly works fine

File: server/core_wallet.go (hunk 2 above)
Comment (ftkg) [reply server/core_wallet.go:211]: `uuid.NewV4()` actually returns `(UUID, error)`

**Comment by ftkg** [comment]
Replaced `pgx.Batch` with `INSERT INTO .. SELECT unnest()` to avoid having to sort input data (due to deadlock potential). The single statement may also be more efficient.

**Comment by redbaron** [comment]
> Replaced `pgx.Batch` with `INSERT INTO .. SELECT unnest()` to avoid having to sort input data (due to deadlock potential). The single statement may also be more efficient.

Queries in Batch are executed in the same order they were added to Batch, which is in turn the same as in unnest, so deadlock potential should be the same.

As for efficiency, pgx uses prepared statement underndeath, so any request after first will be sending just args for the query, not the query itself. `pgx` will spend time hashinb query string (to match  prepared statement), but I think simplicity of Batch makes it more appealing rather than fairly unconventional `unnest` .

If string hashing becomes a problem, then explicit prepared statement can be used with Batch.

**Comment by ftkg** [comment]
> > Replaced `pgx.Batch` with `INSERT INTO .. SELECT unnest()` to avoid having to sort input data (due to deadlock potential). The single statement may also be more efficient.
> 
> Queries in Batch are executed in the same order they were added to Batch, which is in turn the same as in unnest, so deadlock potential should be the same.
> 
> As for efficiency, pgx uses prepared statement underndeath, so any request after first will be sending just args for the query, not the query itself. `pgx` will spend time hashinb query string (to match prepared statement), but I think simplicity of Batch makes it more appealing rather than fairly unconventional `unnest` .
> 
> If string hashing becomes a problem, then explicit prepared statement can be used with Batch.

The only other difference I see is tracking stats for every single query (Batch) vs the single bulk query (unnest) but I don't really prefer one or the other, wdyt?

**Review by zyro — APPROVED** [review]

**Pull request merged**